quickwit/quickwit-search/src/leaf.rs

```diff
@@ -1049,6 +1075,8 @@ impl CanSplitDoBetter {
         // Calculate the number of splits which are guaranteed to deliver enough documents.
         let min_required_splits = splits
             .iter()
+            // splits are sorted by whether they are contained in the request time range
+            .filter(|split| Self::is_contained(split, &request))
             .map(|split| split.num_docs)
             // computing the partial sum
             .scan(0u64, |partial_sum: &mut u64, num_docs_in_split: u64| {
```

Comment:
Can you explain why this works? Say you have 5 splits:
- split_1 is contained in the request 
- split_2 to split_5 are only overlapping the request

In that case `min_required_splits` would be 1 here, but there might be better results in any of the other splits.

Reply:
My thoughts were that there's a

```rust
.take_while(|partial_sum| *partial_sum < num_requested_docs)
```

so `min_required_splits` would not get to be 1, but now I see that I should validate that we actually reached this condition, and only if we reached it to set `min_required_splits` and optimize, otherwise return early.

Reply:
I pushed a quick fix, I need to test it, but I think it answers the problem.

Reply:
I didn't know of `std::ops::ControlFlow` 😄. Nevertheless, this reaches the limit when iterators stop being practical. A `for` loop is much more readable here.